lucene/s3directory/src/test/org/apache/lucene/store/s3/TestS3Directory.java

```
@@ -0,0 +1,342 @@
+      throws IOException {
+    final IndexWriterConfig config = new IndexWriterConfig(analyzer);
+    config.setOpenMode(IndexWriterConfig.OpenMode.CREATE);
+    config.setUseCompoundFile(useCompoundFile);
+
+    try (IndexWriter writer = new IndexWriter(directory, config)) {
+      for (final Object element : docs) {
+        final Document doc = new Document();
+        final String word = (String) element;
+        doc.add(new StringField("keyword", word, Field.Store.YES));
+        doc.add(new StringField("unindexed", word, Field.Store.YES));
+        doc.add(new StringField("unstored", word, Field.Store.NO));
+        doc.add(new StringField("text", word, Field.Store.YES));
+        writer.addDocument(doc);
+      }
+      // FIXME: review
+      // writer.optimize();
+    } catch (Exception e) {
+      logger.log(Level.SEVERE, null, e);
+    }
+  }
+
+  private long timeIndexWriter(final Directory dir) throws IOException {
+    final long start = System.nanoTime();
+    addDocuments(dir, openMode, useCompoundFile, docs);
+    final long stop = System.nanoTime();
+    return stop - start;
+  }
+
+  private IndexWriterConfig getIndexWriterConfig() {
+    final IndexWriterConfig config = new IndexWriterConfig(analyzer);
+    config.setOpenMode(IndexWriterConfig.OpenMode.CREATE_OR_APPEND);
+    return config;
+  }
+
+  @Test
+  public void testSearch() throws IOException, ParseException {
+    try (IndexWriter iwriter = new IndexWriter(s3Directory, getIndexWriterConfig())) {
+      final Document doc = new Document();
+      final String text = "This is the text to be indexed.";
+      doc.add(new Field("fieldname", text, TextField.TYPE_STORED));
+      iwriter.addDocument(doc);
+      if (iwriter.hasUncommittedChanges()) {
+        iwriter.commit();
+      }
+      if (iwriter.isOpen()) {
+        iwriter.getDirectory().close();
+      }
+      iwriter.forceMerge(1, true);
+    } catch (Exception e) {
+      logger.log(Level.SEVERE, null, e);
```

likewise here, let's not swallow exceptions in general